### DISCUSSION TITLE
Artwork file is not found in PDF rendering process since dita-ot-4.1

### DISCUSSION THREAD

**Discussion by T-Spark** [dita-ot/dita-ot#4453] [Q&A]
Hi folks,
I´am using dita-ot for creating and rendering technical documentation in PDF. Now its time to upgrade my dita-ot, I thought and upgraded from 4.0.1 to 4.1. All worked as expected except for one thing: The artwork file which should appear on the cover page didn´t. No idea why. Maybe any of you guys knows why?
The logfile points to the front-matter.xsl file:
```
transform.fo2pdf.fop:
      [fop] [ERROR] Image not found. URI: file:..\images\Titelbild.svg. (See position 16:26431)
```
This is the relevant position in topic.fo:
![Screenshot 2024-03-28 125842](https://github.com/dita-ot/dita-ot/assets/30492005/fd11d7d3-a4be-42de-a98a-b8d8263039a0)
The code in the front-matter.xsl looks like this:
```
<fo:flow flow-name="xsl-region-body">
                <fo:block xsl:use-attribute-sets="__frontmatter">
                    <fo:block-container xsl:use-attribute-sets="__frontmatter__logo__container">
                        <fo:block>
                           <fo:external-graphic src="url(Customization/OpenTopic/common/artwork/Logo.svg)" content-width="53mm"/>
                        </fo:block>
                    </fo:block-container>
                    <!-- set the title -->
                    <fo:block xsl:use-attribute-sets="__frontmatter__title">
                        <xsl:value-of select="$LLdocType"/>
                    </fo:block>
                    <fo:block xsl:use-attribute-sets="__frontmatter__device">
                     <xsl:value-of select="$productName"/><xsl:text> </xsl:text>
                    </fo:block>
                    <fo:block xsl:use-attribute-sets="__frontmatter__producttype">
                        <xsl:value-of select="$LLproductType"/>
                    </fo:block>
                    <fo:block-container xsl:use-attribute-sets="__frontmatter__title__image__container">
                        <fo:block>
                           <fo:external-graphic src="url(file:..\images\Titelbild.svg)" content-width="160mm"/>
                        </fo:block>
                    </fo:block-container>                            
                    <fo:block-container xsl:use-attribute-sets="__frontmatter__version__container">
						<fo:block>
                        <xsl:value-of select="$LLdocEdition"/><xsl:text> </xsl:text>
                        <xsl:value-of select="$LLproductVersion"/><xsl:text> - </xsl:text>
                        <xsl:value-of select="$LLproductVersionDate"/>
						</fo:block>
                    </fo:block-container>
                </fo:block>
```

The relevant line is `<fo:external-graphic src="url(file:..\images\Titelbild.svg)" content-width="160mm"/>`
Must have something to do with the relative path. But, as said, it worked well since a lot of dita-ot versions. Something must have changed regarding the starting point for the path?

Hopefully anyone can help. Thanks in advance and happy easter.
Regards,
Lutz

**Comment by stefan-jung** [comment]
@T-Spark, your URL is invalid. This is the reason, why the file cannot be found.

**Comment by T-Spark** [comment]
Hi stefan-jung,
thanks for your reply. Yes, so far it is clear, that the file cannot be found, but I used this code for years with previous versions of dita-ot and it worked. So why now? What has changed in dita-ot 4.1?

In view of the fact that the content of the FO file (topic.fo) is the same as always, I think that the change must be on the side of the FOP, in this case the Apache FOP.

**Comment by T-Spark** [comment]
Maybe the way the relative path worked was a bug. I don´t know why, but it worked.
For now, I found a workaround by copying the image file into the artwork folder of the dita-ot tree. This is done by the same batch file that calls the dita command:
```
REM Get the absolute path of DITAOT's home directory
set DITA_DIR=C:\DITA-OT\dita-ot-4.1.2
xcopy "..\images\Titelbild.svg" "%DITA_DIR%\plugins\com.novexx.pdf-Doc_oL\cfg\common\artwork" /Y
```
The image file is then processed as follows:
```
<fo:block-container xsl:use-attribute-sets="__frontmatter__title__image__container">
          <fo:block>
                   <fo:external-graphic src="url(Customization/OpenTopic/common/artwork/Titelbild.svg)" content-width="160mm"/>
           </fo:block>
 </fo:block-container> 
```
This not a very elegant solution, but it works. The file "Titelbild.svg" is overwritten for each other manual version with a different title image.

**Comment by stefan-jung** [comment]
@T-Spark, could you please close this issue?
